File: Python/CS50P/C/hogwarts.c
#include <stdio.h>

int main(){
  struct student {
    char *name;
    char *house;
    char *patronus;
  };

  struct student harry = {"Harry", "Gryffindor", "Stag"};
  struct student hermione = {"Hermione", "Gryffindor", "Otter"};
  struct student ron = {"Ron", "Gryffindor", "Jack Russel Terrier"};
  struct student draco = {"Draco", "Slytherin", '\0'};

  printf("%s >> %s >> %s\n", harry.name, harry.house, harry.patronus);
  printf("%s >> %s >> %s\n", hermione.name, hermione.house, hermione.patronus);
  printf("%s >> %s >> %s\n", ron.name, ron.house, ron.patronus);
  printf("%s >> %s >> %s\n", draco.name, draco.house, draco.patronus);

  return 0;
}

File: Python/CS50P/hogwarts.py
# # students = ["Hermione", "Harry", "Ron"]
# # 
# # # for student in students:
# # #     print(student)
# # 
# # for i in range(len(students)):
# #     print(i + 1, students[i])
# 
# students = {"Hermione": "Gryffindor",
#             "Harry": "Gryffindor",
#             "Ron": "Gryffindor",
#             "Draco": "Slytherin",
# }
# 
# # print(students["Hermione"])
# # print(students["Harry"])
# # print(students["Ron"])
# # print(students["Draco"])
# 
# for student in students:
#     print(student, students[student], sep=": ")

students = [
        {"name": "Hermione", "house": "Gryffindor", "patronus": "Otter"},
        {"name": "Harry", "house": "Gryffindor", "patronus": "Stag"},
        {"name": "Ron", "house": "Gryffindor", "patronus": "Jack Russel Terrier"},
        {"name": "Draco", "house": "Slytherin", "patronus": None}
]

for student in students:
    print(student["name"],student["house"], student["patronus"],sep=" - ")
